Verändern der RenderConfiguration per GUI ermöglichen

bakoe opened the pull request:
fixes #23

bakoe committed:
WIP: Add possibility to change the road color using a color picker

bakoe committed:
WIP: Extend render configuration GUI to allow modifying different types and attributes

lheilmann committed:
WIP: Extend GUI to allow text inputs and boolean checkboxes

bakoe committed:
#23: Fix layout of the controls in RenderConfigurationBrowser

bakoe committed:
Merge branch 'dev' into feature/23-veraendern-der-renderconfiguration-per-gui-ermoeglichen

birneamstiel commented:
Wenn ich einfach nur die Farbe von `humanUsedLand` änder, ergibt sich folgendes Bild:

<img width="602" alt="screen shot 2017-06-23 at 11 31 26" src="https://user-images.githubusercontent.com/5410949/27476061-980a3760-5807-11e7-86c3-481f45e740c8.png">

Ist das so gedacht? Scheinbar liegen dann auch mehrere Straßennamen übereinander und die Karte lässt sich nur noch recht schwerfällig bewegen. Bin mir nicht so ganz sicher woran das liegt, aber vielleicht könnt ihr mal schauen ob ihr das reproduziert kriegt @bakoe @lheilmann

bakoe committed:
#23: Fix bug where RenderObjects duplicated themselves while re-rendering

The bug was caused by #13 (Markieren von Orten): There, the call to PolygonMorph's initialization method "vertices: ... color: ... borderWidth: ... borderColor: ..." (which has quite important side effects -- look it up!) within OSMWayRenderObjects's method "renderOn:" was replaced with stand-alone calls to PolygonMorph's setter methods "setVertices:", "color:", "borderWidth:" and "borderColor:".

bakoe commented:
Der von @birneamstiel beschriebene Bug wurde durch die Veränderung der `renderOn:`-Methode der Klasse *OSMWayRenderObject* in #13 verursacht – vor Mergen des *dev*-Branches in den hier genutzten Feature-Branch (92f941c) trat der Bug noch nicht auf.

Der Fehler wurde mit fa4be0a behoben.

birneamstiel commented on packages/OpenSqueakMap.package/OSMRenderConfigurationBrowser.class/instance/color.st:
```diff
@@ -0,0 +1,5 @@
+actions
+color
+
+	self selectedAttribute: #color;
```
minor thing: this way of remembering the selected attribute leads to not being able to edit two different color attributes at once because both of the color pickers will change the same value then. Not sure if this is desired behavior but nevertheless not a big issue...

bakoe replied:
resolved in a2e69aa

birneamstiel commented on packages/OpenSqueakMap.package/OSMRenderConfigurationBrowser.class/instance/setCurrentValueFor.To..st:
```diff
@@ -0,0 +1,14 @@
+	rule basicAttributes at: anAttribute put: aValue.
+	renderConfiguration at: self selectedType put: rule.
+	
+	self viewport chunkManager 
```
Maybe the `RenderConfigurationBrowser` should only know about the `Viewport`, not the `ChunkManager`?

We could add something like `updateRenderConfig` to `Viewport` where the whole updating, applying and invalidating happens

#58

bakoe replied:
resolved in 68d0c40

bakoe committed:
#23: Loose RenderConfigurationBrowser's coupling to its Viewport's ChunkManager

bakoe committed:
#23: Prevent the user from opening multiple colorPickers in the RenderConfiguration GUI

bakoe committed:
Merge branch 'dev' into feature/23-veraendern-der-renderconfiguration-per-gui-ermoeglichen
